### PR TITLE
Dynamic Nldi Map

### PR DESCRIPTION
**Pull request by Rhipeus**
image

### CODE REVIEW

File: src/DashboardUI/src/components/NldiTab.tsx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    if (nldiData.directions & Directions.Upsteam) {
+      directionFilters.push(["==", ["get", "westdaat_direction"], "Upstream"])
+    }
+    if (nldiData.directions & Directions.Downsteam) {
+      directionFilters.push(["==", ["get", "westdaat_direction"], "Downstream"])
+    }
+
+    let pointsLayer = map?.getLayer('nldi-usgs-points');
+    if (pointsLayer) {
+      map?.setFilter(pointsLayer.id,
```
Comment (bbarber) [inline src/DashboardUI/src/components/NldiTab.tsx:163]: Will any of these filters need to be persisted via the URL, or not for the NLDI stuff? I have a pattern in place to handle some layer visibility filters in my branch if so

File: src/DashboardUI/src/components/SidePanel.tsx [hunk 2]
```diff
@@ -10,10 +11,14 @@ interface SidePanelProps {
...
-
-  var tabComponent = props.currentTab === HomePageTab.WaterRights ? <WaterRightsTab />
-    : props.currentTab === HomePageTab.Aggregations ? <AggregationsTab />
-      : <SiteSpecificTab />;
+  const tabs: Record<HomePageTab, JSX.Element> = {
+    [HomePageTab.WaterRights]: <WaterRightsTab />,
+    [HomePageTab.Aggregations]: <AggregationsTab />,
+    [HomePageTab.SiteSpecific]: <SiteSpecificTab />,
+    [HomePageTab.TempNldi]: <NldiTab />,
+  }
```
Comment (bbarber) [inline src/DashboardUI/src/components/SidePanel.tsx:19]: This is way cleaner. I like it

**Comment by bbarber** [comment]
Would you mind including a screenshot for future UI-related PRs? It helps me visualize the html elements while reviewing :)

**Comment by bbarber** [comment]
Lint errors for: react-hooks/exhaustive-deps

Be warned, these can cause infinite loops if they're a function

**Review by bbarber — APPROVED** [review]

File: src/DashboardUI/src/components/NldiTab.tsx (hunk 1 above)
Comment (Rhipeus) [reply src/DashboardUI/src/components/NldiTab.tsx:163]: The data these filters are based on will (directions, datapoints, lat, long) will need to be persisted in the URL.  I wouldn't think that these specific values would.

Comment (Rhipeus) [reply src/DashboardUI/src/components/NldiTab.tsx:171]: I'll figure out the URL stuff later.

File: src/DashboardUI/src/components/NldiTab.tsx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  }, [layers, map, nldiData.dataPoints, nldiData.directions]);
+
+  return (
+    <>
+      <div className='row'>
+        <div className='col-12'>
+          <h1>NLDI Site Search Tool</h1>
+        </div>
+      </div>
+      <div className="form-group">
```
[the remark this answers is not in the document]
Comment (Rhipeus) [reply src/DashboardUI/src/components/NldiTab.tsx]: Changed

**Pull request merged**